Dockerfile

```diff
@@ -2,54 +2,111 @@
-    composer install --working-dir=/opt/kimai --dev --optimize-autoloader && \
-    /opt/kimai/bin/console doctrine:database:create && \
-    /opt/kimai/bin/console doctrine:schema:create && \
-    /opt/kimai/bin/console doctrine:migrations:version --add --all && \
-    /opt/kimai/bin/console cache:warmup && \
-    chown -R www-data:www-data /opt/kimai/var && \
-    chown www-data:www-data /opt/kimai/vendor/mpdf/mpdf/tmp
-
-WORKDIR /opt/kimai
+        xsl \
+        xml \
+        iconv \
+        bcmath \
+        curl \
+        fileinfo \
+        gettext \
+        xmlrpc \
+        xmlwriter \
+        simplexml \
+        zip \
+        opcache
+
+# Install Xdebug
+ARG XDEBUG=0
+
+RUN set -ex \
+    && if [ "$XDEBUG" = "1" ]; then \
+      mkdir /tmp/build \
+      && curl -sSL https://github.com/xdebug/xdebug/archive/2.9.8.tar.gz | tar -xzC /tmp/build \
+      && cd /tmp/build/xdebug-* \
+      && phpize \
+      && ./configure --enable-xdebug \
+      && make \
+      && make install \
+      && rm -rf /tmp/build \
+      && echo "zend_extension=xdebug.so" > /usr/local/etc/php/conf.d/xdebug.ini \
+      && echo "xdebug.idekey=PHPSTORM" >> /usr/local/etc/php/conf.d/xdebug.ini \
+      && echo "xdebug.remote_enable=1" >> /usr/local/etc/php/conf.d/xdebug.ini \
+      && echo "xdebug.remote_connect_back=1" >> /usr/local/etc/php/conf.d/xdebug.ini \
+      && echo "xdebug.remote_host=host.docker.internal" >> /usr/local/etc/php/conf.d/xdebug.ini \
+      && echo "xdebug.remote_port=9000" >> /usr/local/etc/php/conf.d/xdebug.ini \
+      && echo "xdebug.remote_autostart=1" >> /usr/local/etc/php/conf.d/xdebug.ini \
+      && echo "xdebug.remote_handler=dbgp" >> /usr/local/etc/php/conf.d/xdebug.ini; \
+    fi
+
+WORKDIR /var/www/html
+
+# Install composer
+ENV COMPOSER_MEMORY_LIMIT=-1
+
+RUN set -ex \
+    && curl -sS https://getcomposer.org/composer-stable.phar -o /usr/local/bin/composer \
+    && chmod +x /usr/local/bin/composer
+
+# Install symfony cli
+RUN set -ex \
+    && curl -sS https://get.symfony.com/cli/installer | bash \
+    && mv /root/.symfony/bin/symfony /usr/local/bin/symfony
+
+# Install production php.ini and make sore there is a folder to hold the demo database
+RUN set -ex \
+    && cp /usr/local/etc/php/php.ini-development /usr/local/etc/php/php.ini \
+    && sed -i 's/memory_limit = 128M/memory_limit = 512M/g' /usr/local/etc/php/php.ini \
+    && sed -i 's/;opcache.memory_consumption=128/opcache.memory_consumption=256/g' /usr/local/etc/php/php.ini \
+    && sed -i 's/;opcache.max_accelerated_files=10000/opcache.max_accelerated_files=20000/g' /usr/local/etc/php/php.ini \
+    && sed -i 's/log_errors_max_len = 1024/log_errors_max_len = 8192/g' /usr/local/etc/php/php.ini \
+    && mkdir -p /var/data \
+    && chown -R www-data:www-data /var/www /var/data
+
+USER www-data:www-data
+
+# Create shallow clone of kimai and configure it as dev role
+RUN set -ex \
+    && git clone --depth 1 https://github.com/kevinpapst/kimai2.git . \
```

Comment:
It's actually quite uncommon for docker-containers to clone the repo from remote.
Given that the Dockerfile comes with the repo, the sources are already there
and just need to be moved over with a

    ADD . /PATH/WHERE/GIT/WOULD/CLONE/TO

This also enables to make custom build with additional plugins installed into
your locally checked out repository.

Additionally to that, it enables to do per-branch builds using github actions
that can be pushed to docker hub automatically.

Reply:
Yeah, this is legacy and I just kept it this way. @kevinpapst can decide what he wants.

Pro: cloning a "default" setup: You can spool up the image and have a running copy without any additional cloning, mounting and configuration.

Contra: Image size gets bigger because a code snapshot is included.

You can always decide to mount a different volume over the codes. The thing is: This is for DEV only. So you either spool up an instance for demo/testing (where it would be nice if you do not need to check out and mount the same commit the image was built with) or you mount the sources anyway (overriding the given ones).

Not sure which way to go, so i kept the current one.